File: Application/Factories/UserFactory.cs
using Application.Contracts.Factories;
using Application.Dtos;
using AutoMapper;
using Domain.Aggregates;

namespace Application.Factories
{
    public class UserFactory : Factory<UserDto, User>, IUserFactory
    {
        public UserFactory(IMapper mapper) : base(mapper)
        {
        }

        public override User Create(UserDto dto)
        {
            return new User(
                dto.Email,
                dto.FirstName,
                dto.MiddleName,
                dto.LastName,
                dto.Role
            );
        }
    }
}

File: WebClient/src/app/pages/users/shared/user.factory.ts
import { Injectable } from '@angular/core';
import {User} from './user.model';
import {CrudModelFactory} from '../../../shared/factories/crud-model.factory';

@Injectable({
  providedIn: 'root'
})
export class UserFactory extends CrudModelFactory<User>{
  create(jsonData: any): User {
    return jsonData ? new User(
      jsonData.id,
      jsonData.email,
      jsonData.firstName,
      jsonData.middleName,
      jsonData.lastName,
      jsonData.role,
      jsonData.active
    ) : new User();
  }
}
